rules/msk_topic_config.go

```diff
@@ -450,6 +492,50 @@ func (r *MSKTopicConfigRule) validateTieredStorageEnabled(
 	return nil
 }
 
+func (r *MSKTopicConfigRule) validateTieredStorageNotEnabled(
+	runner tflint.Runner,
+	configKeyToPairMap map[string]hcl.KeyValuePair,
+) error {
+	tieredStoragePair, hasTieredStorageAttr := configKeyToPairMap[tieredStorageEnableAttr]
+
+	if !hasTieredStorageAttr {
+		return nil
+	}
+
+	var tieredStorageVal string
+	diags := gohcl.DecodeExpression(tieredStoragePair.Value, nil, &tieredStorageVal)
+	if diags.HasErrors() {
+		return diags
+	}
+
+	if tieredStorageVal != tieredStorageEnabledValue {
+		return nil
+	}
```

Comment:
:thinking: does this work with `bool`s?

```suggestion
	var tieredStorageEnabled bool
	diags := gohcl.DecodeExpression(tieredStoragePair.Value, nil, &tieredStorageEnabled)
	if diags.HasErrors() {
		return diags
	}

	if !tieredStorageEnabled {
		return nil
	}
```

Reply:
Not sure about this... I wouldn't want it to report diagnosis problems if it fails converting. That's why I went with string comparison

Reply:
I also avoid parsing to bool, as values like "1" is working too, as it is parsed as true.